.github/workflows/deploy.yml

```diff
@@ -0,0 +1,105 @@
+#        deploys a RELEASE with the selected version
+#        updates the project version by incrementing the patch version
+#        commits the version update change to the repository's default branch.
+name: Deploy artifacts with Maven
+on:
+  push:
+    branches: [master]
+  release:
+    types: [published]
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Detect release type
+      run: |-
+        if [[ "${{github.event.ref}}" == "refs/heads/master" ]]; then
+          echo "::set-output name=type::SNAPSHOT"
+        else
+          echo "::set-output name=type::RELEASE"
+        fi
+      id: release-detection
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Import Secrets
+      uses: RichiCoder1/vault-action@v0.10.2
+      with:
+        url: ${{ secrets.VAULT_ADDR }}
+        method: approle
+        roleId: ${{ secrets.VAULT_ROLE_ID }}
+        secretId: ${{ secrets.VAULT_SECRET_ID }}
+        secrets: |
+          common/github.com/actions/camunda NEXUS_PSW | NEXUS_PSW;
+          common/github.com/actions/camunda NEXUS_USR | NEXUS_USR;
+          common/github.com/actions/camunda MAVEN_CENTRAL_GPG_SIGNING_KEY_PASSPHRASE | MAVEN_CENTRAL_GPG_SIGNING_KEY_PASSPHRASE;
+          common/github.com/actions/camunda MAVEN_CENTRAL_GPG_SIGNING_KEY_PUB | MAVEN_CENTRAL_GPG_SIGNING_KEY_PUB;
+          common/github.com/actions/camunda MAVEN_CENTRAL_GPG_SIGNING_KEY_SEC | MAVEN_CENTRAL_GPG_SIGNING_KEY_SEC;
+          common/github.com/actions/camunda MAVEN_CENTRAL_DEPLOYMENT_PSW | MAVEN_CENTRAL_DEPLOYMENT_PSW;
+          common/github.com/actions/camunda MAVEN_CENTRAL_DEPLOYMENT_USR | MAVEN_CENTRAL_DEPLOYMENT_USR;
+    - name: Set up Nexus authentication
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+        server-id: camunda-nexus
+        server-username: NEXUS_USR
+        server-password: NEXUS_PSW
+    - name: Set up Maven Central authentication
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+        server-id: central
+        server-username: MAVEN_CENTRAL_DEPLOYMENT_USR
+        server-password: MAVEN_CENTRAL_DEPLOYMENT_PSW
+        settings-path: ${{ github.workspace }}
+    - name: Cache
+      uses: actions/cache@v1
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+    - if: github.event.release
+      name: Update version in pom.xml (Release only)
+      run: mvn -B versions:set -DnewVersion=${{ github.event.release.tag_name }} -DgenerateBackupPoms=false
+    - name: Run Maven (${{ steps.release-detection.outputs.type }})
+      run: mvn -B clean install
+    - name: Publish package (${{ steps.release-detection.outputs.type }})
+      run: |-
+        echo -e "${MVN_CENTRAL_GPG_KEY_SEC}" gpg -q --allow-secret-key-import --import
```

Comment:
not sure I understand, where does this GPG thing come from?

Reply:
uh, there was actually a pipe missing :see_no_evil: